spec/support/shared_examples/validators/shared_examples_for_advocate_claim_validators.rb

```diff
@@ -97,12 +97,163 @@
 
 RSpec.shared_examples 'advocate claim supplier number' do
   context 'supplier_number' do
-      # NOTE: In reality supplier number is derived from external_user which in turn is validated in any event
-      let(:advocate) { build(:external_user, :advocate, supplier_number: '9G606X') }
+    # NOTE: In reality supplier number is derived from external_user which in turn is validated in any event
+    let(:advocate) { build(:external_user, :advocate, supplier_number: '9G606X') }
+
+    it 'should error when the supplier number does not match pattern' do
+      claim.external_user = advocate
+      should_error_with(claim, :supplier_number, 'invalid')
+    end
+  end
+end
+
+RSpec.shared_examples 'common defendant uplift fees aggregation validation' do |options|
+  let(:fee_type) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][0]) }
+  let(:fee_type_2) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][2]) }
+  let(:uplift_fee_type) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][1]) }
+  let(:uplift_fee_type_2) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][3]) }
```

Comment:
🤔 while this is briefer and reusable it sacrifices clarity

Reply:
I have refactored to attempt to make this clearer, and similar to the existing solution. 

Existing spec functionality:-
    let(:miaph) { create(:misc_fee_type, :miaph) }
    let(:miahu) { create(:misc_fee_type, :miahu) }
    let(:midtw) { create(:misc_fee_type, :midtw) }
    let(:midwu) { create(:misc_fee_type, :midwu) }

proposed change:-
  when 'advocate_claim', 'advocate_hardship_claim'
    let(:fee_type_1) { create(:misc_fee_type, :miaph) }
    let(:fee_type_2) { create(:misc_fee_type, :midtw) }
    let(:uplift_fee_type_1) { create(:misc_fee_type, :miahu) }
    let(:uplift_fee_type_2) { create(:misc_fee_type, :midwu) }